crates/bitwarden-crypto/src/keys/key_id.rs

```diff
@@ -9,7 +9,14 @@ pub(crate) const KEY_ID_SIZE: usize = 16;
 /// A key id is a unique identifier for a single key. There is a 1:1 mapping between key ID and key
 /// bytes, so something like a user key rotation is replacing the key with ID A with a new key with
 /// ID B.
-pub(crate) struct KeyId(uuid::Uuid);
+#[derive(Clone)]
+pub(crate) struct KeyId(Uuid);
+
+impl zeroize::Zeroize for KeyId {
+    fn zeroize(&mut self) {
+        self.0 = Uuid::nil();
+    }
+}
```

Comment:
? Why are we implementing zeroize this way.

Reply:
Hm, so initially I wanted to derize zeroize on drop on the signing key. Since the signing key also contains a keyid (which is not secret), we need to impl Zeroize for the keyid, even though we don't really care about it being zeroized. KeyId contains a UUID, which does not expose access to the contained bytes mutably, so this impl, as far as I understand, simply replaces the contained UUID with a null uuid.

That said, maybe we can just remove all of the zeroize deriving since we have the zeroizing allocator?

Reply:
Implemented just like for private keys now. Ed25519_dalek signing keys already implement zeroize on drop. This also drops implementing zeroizing of the keyids.